core/src/saros/net/xmpp/contact/AddContactHelper.java

```diff
@@ -0,0 +1,148 @@
+package saros.net.xmpp.contact;
+
+import java.util.Objects;
+import java.util.function.BiPredicate;
+import org.apache.log4j.Logger;
+import org.jivesoftware.smack.Connection;
+import org.jivesoftware.smack.XMPPException;
+import saros.SarosConstants;
+import saros.exceptions.OperationCanceledException;
+import saros.net.util.XMPPUtils;
+import saros.net.xmpp.JID;
+import saros.net.xmpp.XMPPConnectionService;
+
+/**
+ * This class is mainly moved code from Eclipse {@code AddContactWizard} to add a contact on the
+ * XMPP Roster.
+ *
+ * <p>TODO This code has space for improvements, but probably update smack first..
+ */
+class AddContactHelper {
```

Comment:
Probably a matter of preference/taste, but I like `AddContactUtility` better. Just a suggestion though. If you prefer the current name, keeping it that way is fine with me as well.

```suggestion
class AddContactUtility {
```